test/elixir_analyzer/exercise_test/common_checks/list_prepend_head_test.exs

```diff
@@ -0,0 +1,36 @@
+defmodule ElixirAnalyzer.Support.AnalyzerVerification.ListPrependHead do
+  use ElixirAnalyzer.ExerciseTest
+end
+
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.ListPrependHeadTest do
+  use ElixirAnalyzer.ExerciseTestCase,
+    exercise_test_module: ElixirAnalyzer.Support.AnalyzerVerification.ListPrependHead
+
+  alias ElixirAnalyzer.Constants
+
+  test_exercise_analysis "Solutions using [h | t] to prepend to list",
+    comments: [] do
+    [
+      defmodule MyModule do
+        def prepend_ok(list) do
+          [:ok | list]
+        end
+
+        def concat_lists() do
+          [1, 2, 3] ++ [4, 5, 6]
+        end
+      end
+    ]
+  end
+
+  test_exercise_analysis "Solutions using [h] ++ t to prepend to list",
+    comments: [Constants.solution_list_prepend_head()] do
+    [
+      defmodule MyModule do
+        def prepend_ok(list) do
+          [:ok] ++ list
```

Comment:
Just for piece of mind, could you add a couple more complex examples? Adding a head to a function that produces a list, a prepend inside of a pipe, things like that.

Reply:
Added in 2bb23cd. Thanks!